topdown/http.go

```diff
@@ -350,6 +351,14 @@ func executeHTTPRequest(bctx BuiltinContext, obj ast.Object) (ast.Value, error)
 		}
 	}
 
+	// If the Host header was set, make sure that the TLS server name
+	// matches it. This ensures that TLS sessions get routed based on
+	// the requested hostname only if it is different to the address in
+	// the URL.
+	if host := req.Header.Get(http.CanonicalHeaderKey("Host")); host != "" {
+		tlsConfig.ServerName = host
+	}
```

Comment:
Up in `addHeaders` we don't add the "Host" header to the request, just setting it on the request object https://github.com/open-policy-agent/opa/blob/5a5d2a42f4a9960e88c914a08c21b4289d45fbb2/topdown/http.go#L150-L155 Is that going to mess with this check? Or does `req.Header.Get` get it from the request struct? 💭 Should we be adding it as both header _and_ specify it on the request?

Reply:
Argh, you are right. I was originally setting the server name from `req.Host` and tested that locally, but changed it since it broke unit tests. I then ran in to go modules issues testing this change :-/

I'll go back to the drawing board on it.